### DISCUSSION TITLE
Genymotion SaaS Feature Request

### DISCUSSION THREAD

**Discussion by KQT2000** [Genymobile/feedback#9] [Genymotion SaaS] [answered]
Hi!

I was asked to post a feature request here by a member of Genymotion's Support Team. If possible, could you please tell me if you are going to implement the ability to save Recipes via Genymotion's HTTP API and, most importantly, if the ability to use Kiosk Mode with Genymotion SaaS instances versus Genymotion's PaaS instances on AWS, Google Cloud, Azure, etc? 

It seems as if it would create a bad user experience for users to browse through the open OS of a device, especially when you are embedding a device into a website for the purpose of just using a single app. It is also curious considering the feature is already implemented in Genymotion PaaS, but not SaaS, so I am having a hard time understanding why features are available in one cloud version of the software, but not the other. Are the two different versions of the software for two different use cases, which is especially odd considering they are so similar?

Thanks in advance!
Kahlil

**Comment by jcleneveu-geny** [comment, marked as answer]
Hello Kahlil :wave: 

The whole Genymotion team hope you have a great day :rocket: 
Thank you for your suggestion and your usage of our product.

I'm JC, the product owner of both Genymotion SaaS & Device Image (PaaS).

> If possible, could you please tell me if you are going to implement the ability to save Recipes via Genymotion's HTTP API

We are currently reworking the recipes mechanism for a better implementation, and we will make sure that our users have access to some public HTTP so you can leverage the solution as its best. You should have an update regarding this before the end of the year for sure.

> Most importantly, is the ability to use Kiosk Mode with Genymotion SaaS instances versus Genymotion's PaaS instances on AWS, Google Cloud, Azure, etc?

The feature is available only using ADB / command line (gmsaas).
You can use:

- Lock:
`adb shell cmd activity kiosk start`

- Unlock:
`adb shell cmd activity kiosk stop`

It is not yet publicly documented (and HTTP APIs are not yet available for that), but it will definitely be in the future.
Please also note that a large majority of the Android version have this feature, but some Android versions may not have the kiosk mode (Android 7.1 & 4.4 for instance).

Please let us know in the [support](https://support.genymotion.com/hc/en-us/requests/new?ticket_form_id=83539) if you're having some trouble kiosking an application using the commands above.

> It seems as if it would create a bad user experience for users to browse through the open OS of a device, especially when you are embedding a device into a website for the purpose of just using a single app. It is also curious considering the feature is already implemented in Genymotion PaaS, but not SaaS, so I am having a hard time understanding why features are available in one cloud version of the software, but not the other. Are the two different versions of the software for two different use cases, which is especially odd considering they are so similar?

You're 100% right, this doesn't seem very logical. Let be honest with you here: the reason between these differences is mainly historical. Around five years ago, we started building 2 cloud products with different product visions & strategy, and different teams. However after a few years, our product strategy is more aligned, & our teams have been merged, and we are doing our best to make all features available on all solutions. It takes some time though to backport everything so be patient, but please be sure that for us the end goal would be to bring all Device Image (PaaS) features to our SaaS.

@KQT2000 please let me know if I answered all your questions, and if you have some more, feel free to ask.
Feedback is valuable so I highly appreciate your assessments.

Take care,
Jc
